File: Selenium/Activities/Activity12.py
from selenium import webdriver
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://training-support.net/selenium/dynamic-controls")
    print("The title of the page is: ", driver.title)
    wait = WebDriverWait(driver, 10)
    text_field = driver.find_element(By.ID, "input-text")
    print("The text field is enabled: ", text_field.is_enabled())
    toggle_button = driver.find_element(By.ID,"toggleInput")
    toggle_button.click()
    print("The text field is enabled: ", text_field.is_enabled())
    text_field.send_keys("Enter Input")
    wait.until(EC.visibility_of_element_located((By.ID,"input-text")))
    print("The text in the input field is: ", text_field.get_attribute("value"))

File: Selenium/Activities/activity12.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class activity12 {
    public static void main(String[] args) {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://www.training-support.net/selenium/dynamic-controls");
        System.out.println("The title of the page is: " +driver.getTitle());
        //Find the text field , Check if the text field is enabled and print it.
        WebElement textField = driver.findElement(By.id("input-text"));
        System.out.println("The field is enabled: " + textField.isEnabled());
        //Click the "Enable Input" button to enable the input field.
        driver.findElement(By.id("toggleInput")).click();
        //Check if the text field is enabled again and print it.
        System.out.println("The field is enabled: " + textField.isEnabled());
        driver.quit();
    }
}
